co-log-core/src/Colog/Core/Severity.hs

```diff
@@ -0,0 +1,24 @@
+module Colog.Core.Severity
+       ( Severity (..)
+       , filterSeverity
+       ) where
+
+import Colog.Core.Action (LogAction (..), cfilter)
+
+-- | Severity for the log messages.
+data Severity
+    -- | Information useful for debug purposes
+    = Debug
+    -- | Normal operational information
+    | Info
+    -- | Important (more than average) but not critical information
+    | Notice
+    -- | General warnings
+    | Warning
+    -- | General errors/severe errors
+    | Error
```

Comment:
What do you think about possible changes to this data type? I want to say that I'm quite happy with current data type. But I would like to discuss with you possible ideas I have in mind (though, not of them are good).

1. Other name than `Severity`? Like `Level` or `LogLevel` or `Verbosity`? I like `Severity` but maybe some other name might fit better.
2. Add more severities? Like `Critical`? Or remove severity (like `Notice`: I personally find hard to distinguish when I should print something with `Info` or `Notice`?
3. Following after last question: Maybe we should add to documentation examples of when users need to use levels? For example, 

    * `Debug` for then you want to show output of function that is important for internal development, not for users. Like, result of SQL query.
    * `Info` for describing general steps: starting application, finished downloading.
    * `Notice` for important message: don't forget to update settings, the size of your local storage becomes bigger than average.
    * `Warning` for non-critical failures: couldn't download icon from some service to display
    * `Error` for exceptional situations: couldn't syncronize accounts.

    Maybe you have better ideas for examples 🙂 
4. What do you think about introducing bidirectional single-letter pattern-synonyms for these constructors? There're couple reasons for having additional pattern-synonyms:
    1. Using fancy feature.
    2. Shorter alternative names: we give users multiple ways to use our data type.
    3. Good alignment. For example instead of:

```haskell
log Debug "Foo"
log Warning "Bar"
```

users can write:

```haskell
log D "Foo"
log W "Bar"
```

If you think, that this is a good idea, you can take inspiration from pattern synonym in `tomland` library:

* https://github.com/kowainik/tomland/blob/db96cd45b3b9f04db7ce96c630b160764775c148/src/Toml/PrefixTree.hs#L80-L85